homeassistant/components/cambridge_audio/media_player.py

```diff
@@ -285,3 +293,35 @@ async def async_set_repeat(self, repeat: RepeatMode) -> None:
         if repeat in {RepeatMode.ALL, RepeatMode.ONE}:
             repeat_mode = CambridgeRepeatMode.ALL
         await self.client.set_repeat(repeat_mode)
+
+    @command
+    async def async_play_media(
+        self, media_type: MediaType | str, media_id: str, **kwargs: Any
+    ) -> None:
+        """Play media on the Cambridge Audio device."""
+
+        if media_type not in {
+            CAMBRIDGE_MEDIA_TYPE_PRESET,
+            CAMBRIDGE_MEDIA_TYPE_AIRABLE,
+            CAMBRIDGE_MEDIA_TYPE_INTERNET_RADIO,
+        }:
+            raise HomeAssistantError(
+                f"Unsupported media type for Cambridge Audio device: {media_type}"
+            )
+
+        if media_type == CAMBRIDGE_MEDIA_TYPE_PRESET:
+            preset_id = int(media_id)
+            preset = None
+            for _preset in self.client.preset_list.presets:
+                if _preset.preset_id == preset_id:
+                    preset = _preset
+            if not preset:
+                raise ValueError(f"Missing preset for media_id: {media_id}")
```

Comment:
Is this an user error or can this happen? Ideally we raise `HomeAssistantError` or `ServiceValidationError`

Reply:
@joostlek This is a user error. ServiceValidationError would be appropriate here.

Reply:
Besides this error (and the fact it needs to be translated), anything else that's not explicitly platinum compliant?

Reply:
The errors could use translations

Reply:
Updated 👍